src/app/complex/complex-results/complex-navigator/table-structure/table-structure.component.ts

```diff
@@ -132,4 +136,59 @@ export class TableStructureComponent implements OnChanges {
     return unique;
   }
 
+  classifyComplexesSimilaritiesV2(complexesList: Element[]) {
+    const comparedComplexes: [Element, Element, number][] = [];
+    const finalList: Element[] = [];
+
+    for (const complex of complexesList) {
+      for (const comparedComplex of complexesList) {
+        // for unique comparison
+        if (complex.complexAC >= comparedComplex.complexAC) {
+          comparedComplexes.push([complex, comparedComplex, this.calculateSimilarity(complex, comparedComplex)]);
+        }
+      }
+    }
+    comparedComplexes.sort((a, b) => b[2] - a[2]); // sorting by similarityScore
+
+    // const complexesOrderedSet = new Set<Element>(); // used to have unique complexes
+    // for (const [complex1, complex2] of comparedComplexes) {
+    //   if (!complexesOrderedSet.has(complex1)) {
+    //     complexesOrderedSet.add(complex1);
+    //   }
+    //   if (!complexesOrderedSet.has(complex2)) {
+    //     complexesOrderedSet.add(complex2);
+    //   }
+    // }
+
+    const complexesOrderedSet = this.uniqueComplexesListOrderedBySimilarity(comparedComplexes);
+    // to be used in the table as a 1D array
+    for (const complex of complexesOrderedSet) {
+      finalList.push(complex);
+    }
+    return finalList;
+  }
+
+  uniqueComplexesListOrderedBySimilarity(complexesListSimilarities: [Element, Element, number][]) {
+    const complexesOrderedSet = new Set<Element>();
+    for (const [complex1, complex2, similarityScore] of complexesListSimilarities) {
+      if (similarityScore !== 0) {
+        // console.log(complex1.complexAC, complex2.complexAC, similarityScore);
+        if (!complexesOrderedSet.has(complex1)) {
+          complexesOrderedSet.add(complex1);
+        }
+        if (!complexesOrderedSet.has(complex2)) {
+          complexesOrderedSet.add(complex2);
+        }
+        for (const [complex3, complex4, similarityScore2nd] of complexesListSimilarities) {
+          if (complex1 === complex3 && similarityScore2nd !== 0) {
+            // console.log(complex3.complexAC, complex4.complexAC, similarityScore2nd);
+            if (!complexesOrderedSet.has(complex4)) {
+              complexesOrderedSet.add(complex4);
+            }
+          }
```

Comment:
Here we could add
```
if (complexesOrderedSet.size === this.complexSearch.elements.length) {
          // All complexes have been added, we can return and stop the loops
          return complexesOrderedSet;
        }
```
To return when we already have all the complexes, so we don't need to keep iterating.